### ISSUE TITLE
HSB not working for Silvercrest HG06104A (Smart LED light strip 2M)

### ISSUE THREAD

**Issue by setrin** [Koenkk/zigbee-herdsman-converters#1800]
I have bought Silvercrest HG06104A (Smart LED light strip 2M) from Lidl.
Support for this device has been already added few days back  (https://github.com/Koenkk/zigbee-herdsman-converters/pull/1797).

Color_temp is working, as well as RGB, but not HSB/HSL/HSV. I have tried fooling around with fromZigbee and toZigbee properties, but without any luck. At this point i have no idea what could be the issue.
Logs:

```
Zigbee2MQTT:debug 2020-11-24 19:15:07: Received MQTT message on 'zigbee2mqtt/lidl_led_stripe/set' with data '{"state": "ON", "color": {"hsb": "250,66,100"}}'
Zigbee2MQTT:debug 2020-11-24 19:15:07: Publishing 'set' 'state' to 'lidl_led_stripe'
Zigbee2MQTT:info  2020-11-24 19:15:07: MQTT publish: topic 'zigbee2mqtt/lidl_led_stripe', payload '{"brightness":254,"color":{"h":360,"s":100,"v":100,"x":0.3989,"y":0.3407},"color_temp":255,"linkquality":94,"state":"ON"}'
```

Also, there is one more issue, but maybe it's intentional. When i try to set color_temp along with rgb, color_temp gets overrided and only rgb is used, even tho LED strip has separate LEDs for RGB, CW and WW.

If someone could provide me some guidance, i will be really thankful.

**Comment by MichielMag** [comment]
Hey, could you replace:
```
{
        fingerprint: [{modelID: 'TS0505A', manufacturerName: '_TZ3000_riwp3k79'}],
        model: 'HG06104A',
        vendor: 'Silvercrest',
        description: 'Smart LED light strip 2M',
        extend: generic.light_onoff_brightness_colortemp_colorxy,
        meta: {applyRedFix: true},
}
```

With 
```
{
        fingerprint: [{modelID: 'TS0505A', manufacturerName: '_TZ3000_riwp3k79'}],
        model: 'HG06104A',
        vendor: 'Silvercrest',
        description: 'Smart LED light strip 2M',
        toZigbee: [tz.on_off, tz.silvercrest_smart_led_string],
        fromZigbee: [fz.on_off, fz.silvercrest_smart_led_string],
        meta: {applyRedFix: true},
}
```

It won't completely work, but I've got a hunch that the HSB will work. If that is the case, you could split up the functionality of those converters so you can take only parts of it. And then figure out how to support the rest.

For your second issue: do you have any way to check if it's intentional? Like, could you use the Lidl Home app + gateway to see if that is able to do both white and colored at once?

LED strips often have addressable LED's that can separately or simultaneously drive both the white and RGB LEDs, but it is really up to the controller that receives the zigbee signals to implement that or not and to offer and endpoint for it.

**Comment by setrin** [comment]
Hey, thank you for you replay.
I tried to use fz.silvercrest_smart_led_string and tz.silvercrest_smart_led_string but it doesn't work.

Unfortunately, i don't have Lidl Gateway so i can't really sniff any packets nor observe the intended behaviour of the LED strip. I have tried to use tz.light_color and tz.light_color_temp separately because in the code i found that if tz.light_color_color_temp is used, only either Color or Color_temp is used from message. Using those two separately i was able to get all LEDs light up for a brief moment, but after transition has ended it was back to either RGB LED or white LEDs lighted up. But this doesn't bother me as much as HSB not working for this device.

I tried to look further into the code and it seems like the device doesn't support `moveToHueAndSaturationAndBrightness` command, only `moveToColor` works. Looks like Tuya TS0505A_led is using very similar controller to this one, but it doesn't work at all after i tried to use `tz.tuya_led_control` converter.

I have no experiences with Zigbee and how the communication actually works and what part of the command is on zigbee2mqtt side and what is on side of the device, but in theory, it should be possible to just convert HSB to RGB+Brightness and use `moveToColor`, right? Is there any downside to it?
